enumerables.rb

```diff
@@ -0,0 +1,166 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |x|
+      yield(self[x])
+    end
+    self
+  end
+
+  def my_each_with_index(value = 0)
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = value
+    size.times do |x|
+      yield(self[x], index)
+      index += 1
+    end
+  end
+
+  def my_select(arg = nil)
+    return to_enum(:my_select) unless block_given? || !arg.nil?
+
+    arr = []
+
+    if !arg.nil?
+      my_each do |x|
+        arr.push(x) if arg(x)
+      end
+    else
+      my_each do |x|
+        arr.push(x) if yield(x)
+      end
+    end
+    arr
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_all?(arg = nil)
+    return false unless block_given? || arg.nil? == false || empty?
+
+    if block_given?
+      my_each do |x|
+        return false unless yield(x)
+      end
+    elsif arg.class == Regexp
+      my_each do |x|
+        return false unless x.match(arg)
+      end
+    elsif arg.class == Numeric
+      my_each do |x|
+        return false unless x.class == arg
+      end
+    else
+      my_each do |x|
+        return false unless arg == x
+      end
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    return true if !block_given? && arg.nil? && !(self.select { |x| x }).empty?
+    return false unless block_given? || !arg.nil?
+
+    if block_given?
+      my_each do |x|
+        return true if yield(x) == true
+      end
+    elsif arg.class == Regexp
+      my_each do |x|
+        return true if x.match(arg)
+      end
+    elsif (arg.is_a? Numeric) || (arg.is_a? String)
+      my_each do |x|
+        return true if arg == x
+      end
+    else
+      my_each do |x|
+        return true if x.class == arg
+      end
+      false
+    end
+  end
+
+  # rubocop:enable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_none?(arg = nil, &block)
+    !my_any?(arg, &block)
+  end
+
+  def my_count(arg = nil)
+    counter = 0
+    if arg.nil? == false
+      my_each do |x|
+        counter += 1 unless x != arg
+      end
+    elsif block_given? == true
+      my_each do |x|
+        counter += if yield(x) == true
+                     1
+                   else
+                     0
+                   end
+      end
+    else
+      my_each do |_x|
+        counter += 1
+      end
+    end
+    counter
+  end
```

Comment:
# my_none?
1. [ ]  Returns true if the block never returns true for all elements
### Test 👉 `array.my_none?(&true_block)`

|Expected|Got|
|:---:|:---:|
|`true`|`false`| 
#
2. [ ] Returns false if the block ever returns true for all elements:: range
### Test 👉 `array.my_none?(&true_block)`

Resulted in *NoMethodError*
#
3. [ ] When a Regex is passed as an argument returns true if none of the collection matches the Regex
### Test 👉 `words.my_none?(/z/)`

|Expected|Got|
|:---:|:---:|
|`true`|`false`| 
#
4. [ ] When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
### Test 👉 `words.my_none?(5)`
|Expected|Got|
|:---:|:---:|
|`true`|`false`|